### ISSUE TITLE
can we have a selection-changed event ?

### ISSUE THREAD

**Issue by atahrijouti** [DataTables/Select#32]
As it is right now, this extension offers only `select` and `deselect` events that reflect a selection change, and it can be a bit bothersome if you are only interested in the selection change and not which event.type was fired.

The only workaround right now is to debounce the event handler on both the `select` and `deselect` and hope that they have already fired and are done.

Could we have an event that triggers after all changes to the selection have taken place ?

thanks

**Comment by DataTables** [comment]
At the moment I work around this by using `table.on( 'select delselect', function .... );`. It does mean that the event handler triggers twice though. This event would remove the need for that and it does sound like a good idea because of that. The downside is that there are different code paths for each (consider for example API row selection / deselection versus end user mouse click or keyboard access), so it would require some thought about how to best do this.

**Comment by atahrijouti** [comment]
I agree with the fact that it is a different concept than just probing which rows were selected or deselected. The use-case I would have loved to have a `selection-changed` event is when I am only interested in getting the rows that are selected after all selection altering has finished. Which in my React app, reflect on the UI depending on the, as an example, `selection.length` as a condition to disable a button, or takes the IDs from rows data to update another portion of the view. 
Triggering state change multiple times can affect performance and/or even get me different results from what is really wanted. Like I said in the description of the issue, the only was to :
`table.on( 'select delselect', _debounce(function(){}, 100);`

**Comment by Zavster** [comment]
Any update on this one?

**Comment by DataTables** [comment]
No sorry - it is still in the same state as I don't really know how this could be implemented without using both `select` and `deselect` events (which would thus require an async timer), in which case use the code @abderrahmane-tj suggested above.

The key thing is that a selection changed event could be triggered by, selecting a row (when none was selected before), selecting a different row from what was selected before, adding or removing a row to a current selection or deselecting all rows. Thus the need to use both `select` and `deselect`.

**Comment by AllanJard** [comment]
I've just committed a debounce function to DataTables 2 (currently on a branch, but will merge to master sometime soon, once all of the extensions have been updated). With that, `select` and `deselect` listeners calling the debounced function would be the way to do it. I'm not going to put that into Select just yet, but it is possible I might in future. For now with the debounce it is just a few external lines.
